### PR TITLE
Support 'Add nuget package' for 'Add missing reference'.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/14845

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/Core/Implementation/Suggestions/FixAllGetFixesService.cs [hunk 1]
```diff
@@ -163,8 +164,9 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             }
         }
 
-        private IEnumerable<CodeActionOperation> GetNewFixAllOperations(IEnumerable<CodeActionOperation> operations, Solution newSolution, CancellationToken cancellationToken)
+        private ImmutableArray<CodeActionOperation> GetNewFixAllOperations(IEnumerable<CodeActionOperation> operations, Solution newSolution, CancellationToken cancellationToken)
         {
+            var result = ArrayBuilder<CodeActionOperation>.GetInstance();
```
Comment (jmarolf) [inline src/EditorFeatures/Core/Implementation/Suggestions/FixAllGetFixesService.cs]: >ArrayBuilder [](start = 25, length = 12)

is this superior to `ImmutableArray<CodeActionOperation>.Builder`?

File: src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs [hunk 2]
```diff
@@ -109,6 +109,82 @@ public SymbolSearchUpdateEngine(ISymbolSearchLogService logService)
...
+                    if (symbol.Type == SymbolType.Assembly)
+                    {
+                        result.Add(new PackageWithAssemblyResult(
+                            symbol.PackageName.ToString(),
+                            database.GetPackageVersion(symbol.Index).ToString(),
+                            GetRank(symbol)));
+                    }
+                }
+            }
+#endif
```
Comment (jmarolf) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs]: can you remove this code if it is not going to be used?

**Comment by CyrusNajmabadi** [comment]
Yup. Array builder is pooled

On Nov 1, 2016 7:18 PM, Jonathon Marolf <notifications@github.com> wrote:

@jmarolf commented on this pull request.

________________________________

In src/EditorFeatures/Core/Implementation/Suggestions/FixAllGetFixesService.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Froslyn%2Fpull%2F14860%23pullrequestreview-6740190&data=02%7C01%7Ccyrusn%40microsoft.com%7C98424fd7b2414c0cc32d08d402c695c0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636136499335886986&sdata=kdE%2BDSuie%2BZQg3jsLLlyuIKS2kf%2FoGpiMCgzZiMeCFM%3D&reserved=0>:

>          {
+            var result = ArrayBuilder<CodeActionOperation>.GetInstance();


ArrayBuilder

is this superior to ImmutableArray<CodeActionOperation>.Builder?

-

File: src/EditorFeatures/Core/Implementation/Suggestions/FixAllGetFixesService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Suggestions/FixAllGetFixesService.cs]: Yup.  ArrayBuilders are pooled.

**Pull request merged**